It seems something in the fancy log formatting is eating it up. When I look at the raw logs (click on the gear in the upper-right-hand corner) and search for "error:" I see only the hint at the error, not the error itself. Is there a way to replicate this locally?

```
2021-12-21T17:13:08.0218068Z   INFO: CCompilerOpt.cache_flush[842] : write cache to path -> D:\a\numpy\numpy\build\temp.win32-3.8\ccompiler_opt_cache_clib.py
2021-12-21T17:13:08.0301369Z   error: Command "C:\ProgramData\Chocolatey\bin\gfortran.exe -Wall -g -Wall \
    -g -shared -m32 -Wl,--whole-archive D:\a\numpy\numpy\openblas\openblas.a -Wl,--no-whole-archive \
    -LC:\ProgramData\Chocolatey\lib\mingw\tools\install\mingw32\lib\gcc\i686-w64-mingw32\7.3.0 \
    -LC:\cibw\python\pythonx86.3.8.10\tools\libs -LC:\cibw\python\pythonx86.3.8.10\tools\PCbuild\win32 \
    -Lbuild\temp.win32-3.8 -o build\temp.win32-3.8\Release\.libs\libopenblas.67LVI5PIPIPYANQC3VAWDE6KZKQBFZON.gfortran-win32.dll \
    -Wl,--allow-multiple-definition \
    -Wl,--output-def,build\temp.win32-3.8\Release\libopenblas.67LVI5PIPIPYANQC3VAWDE6KZKQBFZON.gfortran-win32.def \
    -Wl,--export-all-symbols -Wl,--enable-auto-import -static -mlong-double-64" \
    failed with exit status 1
2021-12-21T17:13:08.0644930Z   Building wheel for numpy (pyproject.toml): finished with status 'error'
2021-12-21T17:13:08.0649935Z   ERROR: Failed building wheel for numpy
2021-12-21T17:13:08.0662073Z Failed to build numpy
2021-12-21T17:13:08.0663817Z ERROR: Failed to build one or more wheels
2021-12-21T17:13:08.0673482Z Exception information:
2021-12-21T17:13:08.0674069Z Traceback (most recent call last):
2021-12-21T17:13:08.0675174Z   File "C:\cibw\python\pythonx86.3.8.10\tools\lib\site-packages\pip\_internal\cli\base_command.py", line 164, in exc_logging_wrapper
2021-12-21T17:13:08.0676012Z     status = run_func(*args)
2021-12-21T17:13:08.0676825Z   File "C:\cibw\python\pythonx86.3.8.10\tools\lib\site-packages\pip\_internal\cli\req_command.py", line 205, in wrapper
2021-12-21T17:13:08.0677666Z     return func(self, options, args)
2021-12-21T17:13:08.0678676Z   File "C:\cibw\python\pythonx86.3.8.10\tools\lib\site-packages\pip\_internal\commands\wheel.py", line 175, in run
2021-12-21T17:13:08.0679687Z     raise CommandError("Failed to build one or more wheels")
2021-12-21T17:13:08.0680876Z pip._internal.exceptions.CommandError: Failed to build one or more wheels
```

--- a/.github/workflows/wheels.yml
+++ b/.github/workflows/wheels.yml
@@ -50,10 +50,11 @@ jobs:
       matrix:
         # Github Actions doesn't support pairing matrix values together, let's improvise
         # https://github.com/github/feedback/discussions/7835#discussioncomment-1769026
-        buildplat: 
+        buildplat:
         - [ubuntu-20.04, manylinux_x86_64]
         - [macos-10.15, macosx_*]
         - [windows-2019, win_amd64]
+        - [windows-2019, win32]
         python: ["cp38", "cp39", "cp310"]
         include:
         # manylinux pypy builds
@@ -63,13 +64,14 @@ jobs:
         # MacOS PyPy builds
         # Disabled for now because of a PyPy bug
         # that prevents successful compilation
-        #- buildplat: [macos-10.15, macosx_x86_64] 
+        #- buildplat: [macos-10.15, macosx_x86_64]
         #  python: "pp38"
 
         # Windows PyPy builds
         - buildplat: [windows-2019, win_amd64]
           python: "pp38"
-
+    env:
+      IS_32_BIT: ${{ matrix.buildplat[1] == 'win32' }}
     steps:
       - name: Checkout numpy
         uses: actions/checkout@v2
@@ -80,11 +82,18 @@ jobs:
           # A shallow clone can work when the following issue is resolved:
           # https://github.com/actions/checkout/issues/338
           fetch-depth: 0
-
+      - name: Configure mingw for 32-bit builds
+        run: |
+          # Force 32-bit mingw
+          choco uninstall mingw
+          choco install -y mingw --forcex86 --force --version=7.3.0
+          echo "C:\ProgramData\chocolatey\lib\mingw\tools\install\mingw32\bin" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
+          refreshenv
+        if: ${{ env.IS_32_BIT == 'true' }}
       - name: Build wheels
         uses: pypa/cibuildwheel@v2.3.0
         env:
-          NPY_USE_BLAS_ILP64: 1
+          NPY_USE_BLAS_ILP64: ${{ env.IS_32_BIT == 'true' && '0' || '1' }}
           CIBW_BUILD: ${{ matrix.python }}-${{ matrix.buildplat[1] }}
           CIBW_MANYLINUX_X86_64_IMAGE: manylinux2014
           CIBW_ENVIRONMENT_LINUX: CFLAGS='-std=c99 -fno-strict-aliasing'
@@ -96,6 +105,10 @@ jobs:
                                   OPENBLAS64_=/usr/local
                                   CC=clang
                                   CXX=clang++
+          CIBW_ENVIRONMENT_WINDOWS: OPENBLAS64_=${{ env.IS_32_BIT == 'false' && 'openblas' || '' }}
+                                    OPENBLAS=${{ env.IS_32_BIT == 'true' && 'openblas' || '' }}
+                                    CFLAGS='${{ env.IS_32_BIT == 'true' && '-m32' || '' }}'
+                                    LDFLAGS='${{ env.IS_32_BIT == 'true' && '-m32' || '' }}'
           # TODO: Add universal2 wheels, we need to fuse them manually
           # instead of going through cibuildwheel
           # This is because cibuildwheel tries to make a fat wheel
@@ -103,9 +116,6 @@ jobs:
           # for more info
           CIBW_ARCHS_MACOS: x86_64 arm64
           CIBW_TEST_SKIP: "*_arm64 *_universal2:arm64"
-          # Hardcode for now,blas stuff needs changes for 32-bit
-          CIBW_ENVIRONMENT_WINDOWS: NPY_USE_BLAS_ILP64=1
-                                    OPENBLAS64_=openblas
           CIBW_BUILD_VERBOSITY: 3
           CIBW_BEFORE_BUILD: bash {project}/tools/wheels/cibw_before_build.sh {project}
           CIBW_BEFORE_TEST: pip install -r {project}/test_requirements.txt

--- a/tools/wheels/cibw_before_build.sh
+++ b/tools/wheels/cibw_before_build.sh
@@ -17,8 +17,6 @@ if [[ $RUNNER_OS == "Linux" || $RUNNER_OS == "macOS" ]] ; then
     basedir=$(python tools/openblas_support.py)
     cp -r $basedir/lib/* /usr/local/lib
     cp $basedir/include/* /usr/local/include
-    # TODO: don't copy directories if not arm64,
-    # but we need a way to know if cibuildwheel is cross-compiling arm64 to do this
     if [[ $RUNNER_OS == "macOS" && $PLATFORM == "macosx-arm64" ]]; then
         sudo mkdir -p /opt/arm64-builds/lib /opt/arm64-builds/include
         sudo chown -R $USER /opt/arm64-builds
@@ -46,9 +44,8 @@ if [[ $RUNNER_OS == "macOS" ]]; then
     hdiutil attach -mountpoint /Volumes/gfortran gfortran.dmg
     sudo installer -pkg /Volumes/gfortran/gfortran.pkg -target /
     otool -L /usr/local/gfortran/lib/libgfortran.3.dylib
-    
+
     # arm64 stuff from gfortran_utils
-    # TODO: figure out a way not to do this on x86_64, see above comment for openblas
     if [[ $PLATFORM == "macosx-arm64" ]]; then
         source $PROJECT_DIR/tools/wheels/gfortran_utils.sh
         install_arm64_cross_gfortran

